### PR TITLE
Added the test suite

### PR DESCRIPTION
**Pull request by cristian-vijelie**

### CODE REVIEW

File: Makefile.uk [hunk 1]
```diff
@@ -283,6 +342,70 @@ $(LIBNETTLE_DIR)/umac-nh-n.s: $(LIBNETTLE_ASSEMBLY_DEPS) $(LIBNETTLE_DIR)/umac-n
...
+	$(LIBNETTLE_DIR)/testsuite/twofish-test.c \
+	$(LIBNETTLE_DIR)/testsuite/umac-test.c \
+	$(LIBNETTLE_DIR)/testsuite/xts-test.c \
+	$(LIBNETTLE_DIR)/testsuite/yarrow-test.c \
+	$(LIBNETTLE_DIR)/testsuite/testutils.c \
+	$(LIBNETTLE_DIR)/testsuite/testutils.h
+
+
+$(LIBNETTLE_DIR)/testsuite/.deps: $(LIBNETTLE_TEST_SRCS)
+	$(call verbose_cmd,GEN,libnettle: $(notdir $@), cd $(LIBNETTLE_DIR)/testsuite && cp $^ $(LIBNETTLE_DIR) && touch $@)
```
Comment (craciunoiuc) [inline Makefile.uk:407]: I don't really see a reason to move al sources from the `testsuite` folder to the one above, but, if it works it works.

**Review by craciunoiuc — APPROVED** [review]
Hey Cristi,

The commits that add the test suite look pretty good. I did not see anything bad at first glance.

When you send them on the mailing list I will do another pass over everything and I will check if I missed anything. Maybe you can do the same before sending them, so we can avoid the need for a resubmision ;).

Cezar

File: Makefile.uk (hunk 1 above)
Comment (cristian-vijelie) [reply Makefile.uk:407]: Yes, that's the hack I was telling you about. If you look in the sources, they use some headers that are present in the folder above, but are used as being in the current folder. When libnettle builds, it changes the environment. So, to make unikraft build it, I had 3 options: modify every include to be #include "../something.h" via a patch, move the sources where the headers are, or copy the headers where the test sources are. I chose the 2nd option.

File: Config.uk [hunk 2]
```diff
@@ -2,3 +2,9 @@ menuconfig LIBNETTLE
     bool "libnettle: Symmetric and one-way cryptographic library"
     select LIBNEWLIBC
     default y
+
+if LIBNETTLE
```
Comment (craciunoiuc) [inline Config.uk:6]: Add something like this to mount a filesystem:
```
if PLAT_KVM
	select LIBUK9P
	select LIB9PFS
	select LIBVFSCORE_AUTOMOUNT_ROOTFS
	choice LIBVFSCORE_ROOTFS
		default LIBVFSCORE_ROOTFS_9PFS
	endchoice
```

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
Hey again,

I just realized that a filesystem is needed to ensure 100% functionality on KVM. Please add something like what I suggested in the `Config.uk` file. Also, specify this in the readme.

Cezar

File: Makefile.uk [hunk 3]
```diff
@@ -27,7 +27,7 @@ LIBNETTLE-CINCLUDES-y = -I$(LIBNETTLE_DIR)
 # Global flags
 ################################################################################
 LIBNETTLE_CFLAGS = -DHAVE_CONFIG_H -g -O2 -ggdb3 -Wall -W -Wno-sign-compare   \
--Wmissing-prototypes -Wmissing-declarations -Wstrict-prototypes   \
+-Wnomissing-prototypes -Wmissing-declarations -Wstrict-prototypes   \
```
Comment (craciunoiuc) [inline Makefile.uk:30]:
```
gcc: error: unrecognized command line option ‘-Wnomissing-prototypes’; did you mean ‘-Wno-missing-prototypes’?
```

**Comment by craciunoiuc** [comment]
Looks Good!

**Review by craciunoiuc — APPROVED** [review]
Looks good.

**Pull request merged**